src/utilities/gql.ts

```diff
@@ -1,10 +1,188 @@
+  GET_CONFLICTING_ACTIVITIES = 'get_conflicting_activities',
+  GET_EDSL_FOR_SEQ_JSON = 'getEdslForSeqJson',
+  GET_MODEL_EFFECTIVE_ARGUMENTS = 'getModelEffectiveArguments',
+  GET_NON_CONFLICTING_ACTIVITIES = 'get_non_conflicting_activities',
+  GET_SEQUENCE_SEQ_JSON = 'getSequenceSeqJson',
+  GET_USER_SEQUENCE_SEQ_JSON = 'getUserSequenceSeqJson',
+  INSERT_ACTIVITY_DIRECTIVE = 'insert_activity_directive_one',
+  INSERT_ACTIVITY_DIRECTIVE_TAGS = 'insert_activity_directive_tags',
+  INSERT_ACTIVITY_PRESET = 'insert_activity_presets_one',
+  INSERT_CONSTRAINT_DEFINITION = 'insert_constraint_definition_one',
+  INSERT_CONSTRAINT_DEFINITION_TAGS = 'insert_constraint_definition_tags',
+  INSERT_CONSTRAINT_METADATA = 'insert_constraint_metadata_one',
+  INSERT_CONSTRAINT_MODEL_SPECIFICATION = 'insert_constraint_model_specification_one',
+  INSERT_CONSTRAINT_SPECIFICATIONS = 'insert_constraint_specification',
+  INSERT_CONSTRAINT_TAGS = 'insert_constraint_tags',
+  INSERT_EXPANSION_RULE = 'insert_expansion_rule_one',
+  INSERT_EXPANSION_RULE_TAGS = 'insert_expansion_rule_tags',
+  INSERT_MISSION_MODEL = 'insert_mission_model_one',
+  INSERT_PLAN = 'insert_plan_one',
+  INSERT_PLAN_SNAPSHOT_TAGS = 'insert_plan_snapshot_tags',
+  INSERT_PLAN_TAGS = 'insert_plan_tags',
+  INSERT_SCHEDULING_CONDITION_DEFINITION = 'insert_scheduling_condition_definition_one',
+  INSERT_SCHEDULING_CONDITION_DEFINITION_TAGS = 'insert_scheduling_condition_definition_tags',
+  INSERT_SCHEDULING_CONDITION_METADATA = 'insert_scheduling_condition_metadata_one',
+  INSERT_SCHEDULING_CONDITION_TAGS = 'insert_scheduling_condition_tags',
+  INSERT_SCHEDULING_GOAL_DEFINITION = 'insert_scheduling_goal_definition_one',
+  INSERT_SCHEDULING_GOAL_DEFINITION_TAGS = 'insert_scheduling_goal_definition_tags',
+  INSERT_SCHEDULING_GOAL_METADATA = 'insert_scheduling_goal_metadata_one',
+  INSERT_SCHEDULING_GOAL_TAGS = 'insert_scheduling_goal_tags',
+  INSERT_SCHEDULING_SPECIFICATION = 'insert_scheduling_specification_one',
+  INSERT_SCHEDULING_SPECIFICATION_CONDITION = 'insert_scheduling_specification_conditions_one',
+  INSERT_SCHEDULING_SPECIFICATION_CONDITIONS = 'insert_scheduling_specification_conditions',
+  INSERT_SCHEDULING_SPECIFICATION_GOAL = 'insert_scheduling_specification_goals_one',
+  INSERT_SCHEDULING_SPECIFICATION_GOALS = 'insert_scheduling_specification_goals',
+  INSERT_SEQUENCE = 'insert_sequence_one',
+  INSERT_SEQUENCE_TO_SIMULATED_ACTIVITY = 'insert_sequence_to_simulated_activity_one',
+  INSERT_SIMULATION_TEMPLATE = 'insert_simulation_template_one',
+  INSERT_TAG = 'insert_tags_one',
+  INSERT_TAGS = 'insert_tags',
+  INSERT_USER_SEQUENCE = 'insert_user_sequence_one',
+  INSERT_VIEW = 'insert_view_one',
+  MERGE_REQUEST = 'merge_request_by_pk',
+  MERGE_REQUESTS = 'merge_request',
+  MISSION_MODELS = 'mission_model',
+  PLAN = 'plan_by_pk',
+  PLANS = 'plan',
+  PLAN_DATASETS = 'plan_dataset',
+  PLAN_SNAPSHOTS = 'plan_snapshot',
```

Comment:
Not a regression but just noticing this now - can take a plan snapshot even if user does not have edit permissions on a plan. Is this intended?

Reply:
@AaronPlave do you mind going over how to repro this? When I switch to a `user` who didn't create the plan, I see the "Take Snapshot" button disabled.

Reply:
I'm able to repro by making a plan as user A and then switching to user B and looking at plan metadata panel - take snapshot is enabled for me.